wave_lang/kernel/ops/wave_ops.py

```diff
@@ -2360,7 +2360,10 @@ def __post_init__(self):
 
     @property
     def indexing_dims(self) -> list[IndexSymbol]:
-        return self.target_shape
+        from ..wave.utils.general_utils import infer_dim
+
+        dims = [infer_dim(expr) for expr in self.target_shape]
```

Comment:
Can you add a comment explaining why we use infer_dim here? (specifically so that it allows broadcast ops to correctly process symbolic dims of the form binary_op(K, x) where K is a symbol and x is an integer.